src/lib.rs

```diff
@@ -1329,7 +1329,7 @@ impl<I: Clone, O, T: Parser<I, O> + ?Sized> Parser<I, O> for Arc<T> {
         debugger: &mut D,
         stream: &mut StreamOf<I, Self::Error>,
     ) -> PResult<I, O, Self::Error> {
-        debugger.invoke::<_, _, T>(&*self, stream)
+        debugger.invoke::<_, _, T>(self, stream)
```

Comment:
Could you make sure that this isn't a recursive call? Perhaps by doing `&**` instead.